File: backend/app/routers/classify.py
from fastapi import APIRouter, HTTPException, status

from app.schemas import ClassifyImageRequest, ClassifyImageResponse
from app.services.ai_classifier import classify_image

router = APIRouter(tags=["classify"])


@router.post("/classify-image", response_model=ClassifyImageResponse, response_model_by_alias=True)
async def classify_image_endpoint(body: ClassifyImageRequest) -> ClassifyImageResponse:
    if not body.image_base64:
        raise HTTPException(status_code=400, detail="imageBase64 is required")

    try:
        return await classify_image(body.image_base64, body.mime_type)
    except ValueError as exc:
        raise HTTPException(status_code=503, detail=str(exc)) from exc
    except RuntimeError as exc:
        raise HTTPException(status_code=502, detail=str(exc)) from exc

File: backend/src/routes/classify.js
import { Router } from 'express'
import { classifyImage } from '../services/aiClassifier.js'

const router = Router()

/**
 * POST /api/classify-image - Classify infrastructure issue image using OpenAI Vision
 */
router.post('/classify-image', async (req, res, next) => {
  try {
    const { imageBase64, image_base64, mimeType = 'image/jpeg', mime_type = 'image/jpeg' } = req.body

    const base64Data = imageBase64 || image_base64
    const resolvedMimeType = mimeType || mime_type || 'image/jpeg'

    if (!base64Data) {
      return res.status(400).json({ detail: 'imageBase64 is required' })
    }

    const result = await classifyImage(base64Data, resolvedMimeType)
    return res.json(result)
  } catch (err) {
    const statusCode = err.statusCode || 500
    return res.status(statusCode).json({ detail: err.message || 'Classification failed' })
  }
})

export default router
